src/include/rbd/librbd.h

```diff
@@ -407,6 +410,13 @@ CEPH_RBD_API int rbd_mirror_peer_set_client(rados_ioctx_t io_ctx,
 CEPH_RBD_API int rbd_mirror_peer_set_cluster(rados_ioctx_t io_ctx,
                                              const char *uuid,
                                              const char *cluster_name);
+CEPH_RBD_API int rbd_mirror_peer_get_attributes(
+    rados_ioctx_t p, const char *uuid, char *keys, size_t *max_key_len,
+    char *values, size_t *max_value_len);
+CEPH_RBD_API int rbd_mirror_peer_set_attributes(
+    rados_ioctx_t p, const char *uuid, const char *keys, const char *values,
+    size_t key_value_count);
```

Comment:
LGTM, though personally I would prefer if `rbd_mirror_peer_get_attributes` also had and set `size_t &key_value_count` , so it would be symmetrical to `rbd_mirror_peer_set_attributes` and probably a little nicer to work with from C. Or use vectors of strings for keys and values, similarly to what we already do in other API functions, despite of the need to add ugly `rbd_mirror_peer_get_attributes_cleanup`.

But again, just IMHO, and feel free to ignore.

Reply:
Sure, I can add a ```size_t *count``` out-parameter.